munininfluxdb/munin.py

```diff
@@ -1,77 +1,125 @@
 from __future__ import print_function
+from collections import namedtuple
 import os
 import sys
 import pprint
 
 from utils import ProgressBar, Symbol
 from settings import Settings
 
-from vendor import storable
+import storable
 
-def discover_from_datafile(settings):
+DataFileLine = namedtuple(
+    'DataFileLine',
+    'plugin value domain host field property')
+
+
+def parse_datafile_line(line):
```

Comment:
Extracted this block of code as separate function which allows us to test it separately.